components/agents/org.wso2.carbon.identity.entitlement.proxy/src/main/org/wso2/carbon/identity/entitlement/proxy/thrift/ThriftEntitlementServiceClient.java

```diff
@@ -78,7 +78,7 @@ public boolean subjectCanActOnResource(String subjectType, String alias, String
         String xacmlRequest = XACMLRequetBuilder.buildXACML3Request(tempArr);
         EntitlementThriftClient.Client client = getThriftClient(appId);
         Authenticator authenticator = getAuthenticator(serverUrl, userName, password);
-        return (getDecision(xacmlRequest, client, authenticator.getSessionId(false))).contains("Permit");
+        return (getDecision(xacmlRequest, client, authenticator)).contains("Permit");
```

Comment:
if getDecision returns null this would lead to a NPE. Better to avoid the possibility.